src/doc/src/reference/unstable.md

```diff
@@ -1289,6 +1290,96 @@ edition that may break your build.
 
 [edition]: ../../edition-guide/index.html
 
+## Profile `trim-paths` option
+
+* Tracking Issue: [rust-lang/cargo#12137](https://github.com/rust-lang/cargo/issues/12137)
+* Tracking Rustc Issue: [rust-lang/rust#111540](https://github.com/rust-lang/rust/issues/111540)
+
+This adds a new profile setting to control how paths are sanitised in the resulting binary.
+This can be enabled like so:
+
+```toml
+cargo-features = ["trim-paths"]
+
+[package]
+# ...
+
+[profile.release]
+trim-paths = ["all"]
+```
+
+To set this in a profile in Cargo configuration, you need to use either
+`-Z trim-paths` or `[unstable]` table to enable it. For example,
+
+```toml
+# .cargo/config.toml
+[unstable]
+trim-paths = true
+
+[profile.release]
+rustflags = ["diagnostics"]
+```
+
+### Documentation updates
+
+#### trim-paths
+
+*as a new ["Profiles settings" entry](./profiles.html#profile-settings)*
+
+`trim-paths` is a profile setting which enables and controls the sanitisation of file paths in build outputs.
+It takes either string or an array of the following values:
```

Comment:
Not seeing anywhere that we specify an empty list

Reply:
Actually there are only two combinations are necessary:

* `[diagnostics, macro]`
* `[diagnostics, object]`

Other combinations can be represented by just one value.

I wonder if we really want an array for it.

Reply:
So this comment was on the documentation for how the end-user configures things, so we'd likely to keep it as specified.  Assuming you meant to refer to the internal representation, I'd worry its over optimization.  Notice my idea would add more allocations (though less indirection).

Reply:
Not really. I am thinking that the current way to configure trim-paths is a bit confusing. You can have `trim-paths = ["all", "diagnostics", "macro"]` and it is exactly the same as `trim-paths = "all". If we add one more `"none"` to the array it become `trim-paths = "none"`.

We can implement the way as RFC says, but we might want to think about it later.

Reply:
It also tricky to determine the meaning of `trim-paths = []`. Slightly perfer `none` for empty array now.

Reply:
So `none` has higher precedence and wipes everything out?  Thats bizarre.

Reply:
To avoid this issue, I just disallow "all" and "none" presence in the array. However, it makes `serialize` and `deserialize` asymmetric and not match each other.